src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -592,6 +592,11 @@ PALAPI
 PAL_LoadLibraryDirect(
     IN LPCWSTR lpLibFileName)
 {
+    if (!lpLibFileName && g_running_in_exe)
```

Comment:
I think this warrants a comment.

Reply:
There is a helper that is self-explanatory `ShouldRedirectToCurrentLibrary`. Just made the code call that.